oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -647,6 +659,19 @@ public boolean equals(Object obj) {
         && Objects.equals(this.quotaProjectId, other.quotaProjectId);
   }
 
+  private String signJsonWebSignature(JsonFactory jsonFactory, JsonWebSignature.Header header, JsonWebToken.Payload payload) throws IOException {
+    String signedContentString = Base64.encodeBase64URLSafeString(jsonFactory.toByteArray(header)) + "." + Base64.encodeBase64URLSafeString(jsonFactory.toByteArray(payload));
+    byte[] signedContentBytes = StringUtils.getBytesUtf8(signedContentString);
+    try {
+      byte[] signature = this.sign(signedContentBytes);
+      return signedContentString + "." + Base64.encodeBase64URLSafeString(signature);
+
+    } catch (SigningException e) {
```

Comment:
Is there a reason to convert this to an IOException instead of declaring this method to throw SigningException?

Reply:
No, not really, I just kept the pattern that was there before.

SigningException is unchecked. Whilst the api for refreshAccessToken declares that it throws IOException. (I'm not sure I agree that SigningException should be unchecked, but that is outside the scope of this change).

In any case I argue that a failing signature should be handled by the calling code, and as such it should be changed to an IOException before exiting refreshAccessToken

Reply:
@elharo I'll await your answer on this before I reupload. (I'm I doing this correctly with the --amends by the way?)

Reply:
I honestly do not have enough git-fu to tell. Whatever you're doing seems to work fine on my end. Just commit and upload as you need to. We'll squash the commits when we merge.